src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -146,14 +157,26 @@ ss::future<> kafka_client_cache::clean_stale_clients() {
             return now >= (item.last_used + keep_alive);
         };
     };
+
+    gate_guard guard{_gc_gate};
+    auto units = _gc_lock.get_units();
```

Comment:
I don't know if this is really needed, `clean_stale_clients` appears to be reentrancy safe.

It might be worth keeping it and bailing if the lock is already taken:

```suggestion
    auto units = _gc_lock.try_get_units();
    if (!units) {
        co_return;
    }
    gate_guard guard{_gc_gate};
```